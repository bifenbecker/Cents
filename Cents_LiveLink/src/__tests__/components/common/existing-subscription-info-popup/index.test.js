import React from "react";
import {Router as MemoryRouter} from "react-router-dom";
import {createMemoryHistory} from "history";
import {render, fireEvent, act, waitFor} from "@testing-library/react";
import ExistingSubscriptionPopUp from "../../../../components/common/existing-subscription-info-popup";

const history = createMemoryHistory();

const renderExistingSubscriptionPopUp = props => {
  const {getByText} = render(
    <MemoryRouter history={history}>
      <ExistingSubscriptionPopUp {...props} />
    </MemoryRouter>
  );

  return {getByText};
};

describe("Testing ExistingSubscription Component", () => {
  const onIgnoreExistingSubscriptions = jest.fn();
  const toggle = jest.fn();
  const existingSubscriptionsList = [];

  afterEach(() => {
    history.push("/");
  });

  it("Test existing subscription component UI snapshot", async () => {
    expect(
      render(
        <ExistingSubscriptionPopUp
          isOpen
          toggle={toggle}
          existingSubscriptionsList={existingSubscriptionsList}
          onIgnoreExistingSubscriptions={onIgnoreExistingSubscriptions}
        />
      )
    ).toMatchSnapshot();
  });

  it("Start new order button is clicked", async () => {
    const {getByText} = renderExistingSubscriptionPopUp({
      toggle,
      isOpen: true,
      existingSubscriptionsList,
      onIgnoreExistingSubscriptions,
    });
    const button = await getByText("START NEW ORDER");
    fireEvent.click(button);

    expect(onIgnoreExistingSubscriptions).toBeCalled();
  });

  it("Cancel button is clicked", async () => {
    const {getByText} = renderExistingSubscriptionPopUp({
      toggle,
      isOpen: true,
      existingSubscriptionsList,
      onIgnoreExistingSubscriptions,
    });
    const button = await getByText("CANCEL");
    fireEvent.click(button);

    expect(toggle).toBeCalled();
  });

  it("Should provide link to view subscription with appropriate text", async () => {
    const props = {
      toggle,
      isOpen: true,
      onIgnoreExistingSubscriptions,
      existingSubscriptionsList: [1],
    };

    const {getByText: getSingleSubscriptionText} = renderExistingSubscriptionPopUp(props);
    expect(
      await getSingleSubscriptionText("View existing recurring order")
    ).toBeInTheDocument();

    props.existingSubscriptionsList.push([2]);
    const {getByText: getMultipleSubscriptionsText} = renderExistingSubscriptionPopUp(
      props
    );
    expect(
      await getMultipleSubscriptionsText("View existing recurring orders")
    ).toBeInTheDocument();
  });

  it("Should close the modal and redirect to susbcription URL with search parameter when clicked on View Existing Recurring Order", async () => {
    const {getByText} = renderExistingSubscriptionPopUp({
      toggle,
      isOpen: true,
      onIgnoreExistingSubscriptions,
      existingSubscriptionsList: [{recurringSubscriptionId: 1}],
    });
    // Verifying if currently we are not in subscriptions
    expect(history.location.pathname).toEqual("/");

    const viewExistingSubscriptionButton = await getByText(
      "View existing recurring order"
    );
    fireEvent.click(viewExistingSubscriptionButton.parentNode);

    expect(history.location.pathname).toEqual(expect.stringContaining("/subscriptions"));
    expect(history.location.search).toEqual(expect.stringMatching(/subscriptionId=1/));
  });

  it("Should close the modal and redirect to susbcription URL with no search parameter when clicked on View Existing Recurring Orders", async () => {
    const {getByText} = renderExistingSubscriptionPopUp({
      toggle,
      isOpen: true,
      onIgnoreExistingSubscriptions,
      existingSubscriptionsList: [
        {recurringSubscriptionId: 1},
        {recurringSubscriptionId: 2},
      ],
    });
    // Verifying if currently we are not in subscriptions
    expect(history.location.pathname).toEqual("/");

    const viewExistingSubscriptionButton = await getByText(
      "View existing recurring orders"
    );
    fireEvent.click(viewExistingSubscriptionButton.parentNode);

    expect(history.location.pathname).toEqual(expect.stringContaining("/subscriptions"));
    expect(history.location.search).toEqual("");
  });

  it("should set fontSize according to window width", async () => {
    const {getByText} = renderExistingSubscriptionPopUp({
      toggle,
      isOpen: true,
      onIgnoreExistingSubscriptions,
      existingSubscriptionsList: [
        {recurringSubscriptionId: 1},
        {recurringSubscriptionId: 2},
      ],
    });
    const cancelBtnTxt = "CANCEL";
    const newOrderBtnTxt = "CANCEL";

    // width < 300
    act(() => {
      global.innerWidth = 250;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(cancelBtnTxt)).toHaveStyle({fontSize: "12px"}));
    await waitFor(() =>
      expect(getByText(newOrderBtnTxt)).toHaveStyle({fontSize: "12px"})
    );

    // 300 < width < 368
    act(() => {
      global.innerWidth = 340;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(cancelBtnTxt)).toHaveStyle({fontSize: "14px"}));
    await waitFor(() =>
      expect(getByText(newOrderBtnTxt)).toHaveStyle({fontSize: "14px"})
    );

    // width >= 368
    act(() => {
      global.innerWidth = 368;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(cancelBtnTxt)).toHaveStyle({fontSize: "16px"}));
    await waitFor(() =>
      expect(getByText(newOrderBtnTxt)).toHaveStyle({fontSize: "16px"})
    );
  });
});
