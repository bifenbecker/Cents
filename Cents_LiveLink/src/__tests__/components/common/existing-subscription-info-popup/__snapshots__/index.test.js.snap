// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ExistingSubscription Component Test existing subscription component UI snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
      >
        <div
          style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
        />
        <div
          style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
        >
          <div
            style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
          />
          <div
            style="width: 100%; height: 100%; overflow: auto;"
          >
            <div
              class="sc-bczRLJ sc-gsnTZi tmhJc hzPtvw"
            >
              <div
                class="sc-bczRLJ sc-gsnTZi dXYuPw hzPtvw"
              >
                <div
                  class="sc-bczRLJ aQXOc"
                  color="primary"
                  font-weight="700"
                >
                  You have an existing recurring order
                </div>
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi cxsUhz hzPtvw"
              >
                <div
                  class="sc-bczRLJ sc-gsnTZi inRGnP hzPtvw"
                  width="100%"
                >
                  <div
                    class="sc-bczRLJ bTkovy"
                    color="black"
                    font-family="secondary"
                    font-size="18px"
                  />
                </div>
                <div
                  class="sc-bczRLJ sc-gsnTZi hpjsKU hzPtvw"
                  width="100%"
                >
                  <div
                    class="sc-bczRLJ dYIJvR"
                    color="BLACK"
                    font-size="24px"
                    font-weight="900"
                  >
                    Next pickup scheduled for
                  </div>
                  <div
                    class="sc-bczRLJ dYIJvR"
                    color="BLACK"
                    font-size="24px"
                    font-weight="900"
                  />
                </div>
                <div
                  class="sc-bczRLJ sc-gsnTZi inRGnP hzPtvw"
                  width="100%"
                >
                  <div
                    class="sc-bczRLJ iwVVky"
                    color="black"
                    font-family="secondary"
                    font-size="18px"
                  >
                    , 
                  </div>
                </div>
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi jgepW hzPtvw"
              >
                <div
                  class="sc-bczRLJ lhGPIT"
                  color="primary"
                  font-weight="500"
                >
                  View existing recurring order
                  
                </div>
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi cYXLjH hzPtvw"
                width="100%"
              >
                <button
                  class="sc-bczRLJ cQxld"
                  font-family="primary"
                  font-size="16px"
                  height="62px"
                  width="48%"
                >
                  CANCEL
                </button>
                <button
                  class="sc-bczRLJ fAfCNU"
                  font-size="16px"
                  height="62px"
                  width="100%"
                >
                  START NEW ORDER
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
    >
      <div
        style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
      />
      <div
        style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
      >
        <div
          style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
        />
        <div
          style="width: 100%; height: 100%; overflow: auto;"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi tmhJc hzPtvw"
          >
            <div
              class="sc-bczRLJ sc-gsnTZi dXYuPw hzPtvw"
            >
              <div
                class="sc-bczRLJ aQXOc"
                color="primary"
                font-weight="700"
              >
                You have an existing recurring order
              </div>
            </div>
            <div
              class="sc-bczRLJ sc-gsnTZi cxsUhz hzPtvw"
            >
              <div
                class="sc-bczRLJ sc-gsnTZi inRGnP hzPtvw"
                width="100%"
              >
                <div
                  class="sc-bczRLJ bTkovy"
                  color="black"
                  font-family="secondary"
                  font-size="18px"
                />
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi hpjsKU hzPtvw"
                width="100%"
              >
                <div
                  class="sc-bczRLJ dYIJvR"
                  color="BLACK"
                  font-size="24px"
                  font-weight="900"
                >
                  Next pickup scheduled for
                </div>
                <div
                  class="sc-bczRLJ dYIJvR"
                  color="BLACK"
                  font-size="24px"
                  font-weight="900"
                />
              </div>
              <div
                class="sc-bczRLJ sc-gsnTZi inRGnP hzPtvw"
                width="100%"
              >
                <div
                  class="sc-bczRLJ iwVVky"
                  color="black"
                  font-family="secondary"
                  font-size="18px"
                >
                  , 
                </div>
              </div>
            </div>
            <div
              class="sc-bczRLJ sc-gsnTZi jgepW hzPtvw"
            >
              <div
                class="sc-bczRLJ lhGPIT"
                color="primary"
                font-weight="500"
              >
                View existing recurring order
                
              </div>
            </div>
            <div
              class="sc-bczRLJ sc-gsnTZi cYXLjH hzPtvw"
              width="100%"
            >
              <button
                class="sc-bczRLJ cQxld"
                font-family="primary"
                font-size="16px"
                height="62px"
                width="48%"
              >
                CANCEL
              </button>
              <button
                class="sc-bczRLJ fAfCNU"
                font-size="16px"
                height="62px"
                width="100%"
              >
                START NEW ORDER
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
